src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -664,6 +671,31 @@ private static bool ShowInStackTrace(MethodBase mb)
             return !(mb.IsDefined(typeof(StackTraceHiddenAttribute)) || (mb.DeclaringType?.IsDefined(typeof(StackTraceHiddenAttribute)) ?? false));
         }
 
+        private static void ResolveStateMachineMethod(ref MethodBase mb, ref Type declaringType)
+        {
+            Debug.Assert(mb != null);
+            Debug.Assert(declaringType != null);
+
+            Type dt = declaringType.DeclaringType;
+            if (dt != null)
+            {
+                var methods = dt.GetMembers(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance);
```

Comment:
I think `dt.DeclaredMethods` could be slightly cheaper here (don't need non-method members nor methods declared on base types).

Reply:
Is an extension on [TypeInfo rather than Type also IEnumerable](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Reflection/TypeInfo.cs#L24-L31) rather than array. Will add `BindingFlags.DeclaredOnly`

Reply:
And change to `GetMethods`

Reply:
Done